### PR TITLE
front: update front dependencies

### LINKED ISSUE #6555: Update front libs
### Description and goal

Front dependencies haven't been updated since a feew month ago

### Acceptance criteria

- [ ] All outdated dependencies are up to date

### PR DESCRIPTION
**Pull request by SharglutDev**
close #6555 

- prettier 3.* update : add `"trailingComma": "es5"` in `.prettierrc.json` to (almost) keep current behavior (https://prettier.io/docs/en/options.html#trailing-commas)
- eslint updates : add aria-label for buttons without text and create a text if necessary (https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/control-has-associated-label.md) 
- add @rjsf/utils lib to handle rjsf types in the application which aren't available in @rjsf/core anymore (https://rjsf-team.github.io/react-jsonschema-form/docs/api-reference/utility-functions/)
- add @rjsf/validator-ajv8 lib which provides a validator that we can use on @rjsf/core's Form component which now requires the validator prop to check form validity and display errors (https://rjsf-team.github.io/react-jsonschema-form/docs/usage/validation/#api-documentation)
- add @vitest/coverage-v8 which is now required in vite 5.*
- redux dependencies : 
           - replace all `useDispatch` in the app with `useAppDispatch` for better typing (https://redux.js.org/tutorials/typescript-quick-start#define-typed-hooks)
           - replace `createStore()` by `configureStore()` (https://redux.js.org/usage/migrations/migrating-rtk-2#createstore-deprecation)
           - update the fix for a bug in @rtk-query/codegen-openapi (see https://github.com/rtk-incubator/oazapfts/pull/7)
- axios update : add a new customAxios to handle params serialization in get method
- remove RunningTime HOC

There are probably some `import type` missing, they will be handled with https://github.com/osrd-project/osrd/pull/6550

### CODE REVIEW

File: front/src/common/api/axios.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import axios, { AxiosRequestConfig } from 'axios';
+import { mapValues } from 'lodash';
+
+function checkParams(params: Record<string, unknown>) {
```
Comment (anisometropie) [inline front/src/common/api/axios.ts]:
```suggestion
function serializeParams(params: AxiosRequestConfig['params']) {
```

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]
Very good job, thx! LGTM, but waiting for Valentin to review redux part.

File: front/src/applications/editor/components/LinearMetadata/FormComponent.tsx [hunk 2]
```diff
@@ -70,33 +72,10 @@ const IntervalEditorComponent: React.FC<FieldProps> = (props) => {
   const requiredFilter = (requireds: string[]) =>
     requireds.filter((r) => ['end', 'begin'].includes(r));
 
-  // Compute the JSON schema of the linear metadata item
-  const jsonSchema = useMemo(
```
Comment (anisometropie) [inline front/src/applications/editor/components/LinearMetadata/FormComponent.tsx:74]: you no longer need this transformation?

File: front/src/modules/rollingStock/helpers/__tests__/utils.spec.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ function setupEffortCurve(tractionMode: string, max_efforts: number[], speeds: n
   curves.curves[0].curve.speeds = speeds;
   return curves;
 }
-const tMock: TFunction = (key: string) => key;
+const tMock = (key: string) => key;
```
Comment (anisometropie) [inline front/src/modules/rollingStock/helpers/__tests__/utils.spec.ts:13]: why don’t you keep the type here?

File: front/src/modules/simulationResult/components/SpaceTimeChart/createTrain.ts [hunk 4]
```diff
@@ -23,10 +19,8 @@ import { Train, SimulationTrain } from 'reducers/osrdsimulation/types';
  * called with keyValues ['time', 'position']
  */
 export default function createTrain(
-  dispatch: Dispatch,
   keyValues: ChartAxes,
-  simulationTrains: Train[],
-  t: TFunction<Namespace<string>, undefined>
+  simulationTrains: Train[]
```
Comment (anisometropie) [inline front/src/modules/simulationResult/components/SpaceTimeChart/createTrain.ts:21]: update the tsdoc as well

**Review by anisometropie — CHANGES_REQUESTED** [review]
just a few changes.
Overall would have been nice have individual package bump, those that involve changes throughout a lot of files, in separate commits.
But not really worth the time recreating all the commits now.

But if you want to keep the axios commit separate for example, better move all the changes in the same commit, including package.json

anyway good job on this pr !

File: front/src/common/api/axios.ts (hunk 1 above)
Comment (SharglutDev) [reply front/src/common/api/axios.ts]: done

File: front/src/applications/editor/components/LinearMetadata/FormComponent.tsx (hunk 2 above)
Comment (SharglutDev) [reply front/src/applications/editor/components/LinearMetadata/FormComponent.tsx:74]: it's already done in l.437, it seems to be an omission of a previous pr

File: front/src/modules/rollingStock/helpers/__tests__/utils.spec.ts (hunk 3 above)
Comment (SharglutDev) [reply front/src/modules/rollingStock/helpers/__tests__/utils.spec.ts:13]: It seems that the i18next update has changed something in this function type. If you put it back there is this error : 

> Property '$TFunctionBrand' is missing in type '(key: string) => string' but required in type 'TFunction<"translation", undefined>'

The `(key: string) => key;` still works as expected in the test but typescript wants this property and I didn't achieve to add it properly.

File: front/src/modules/simulationResult/components/SpaceTimeChart/createTrain.ts (hunk 4 above)
Comment (SharglutDev) [reply front/src/modules/simulationResult/components/SpaceTimeChart/createTrain.ts:21]: done

**Review by anisometropie — APPROVED** [review]

**Review by nicolaswurtz — APPROVED** [review]

**Pull request merged**